nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/CopyDocumentTest.java

```diff
@@ -0,0 +1,126 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.automation.AutomationService;
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.event.CoreEventConstants;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+import com.google.inject.Inject;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ CoreFeature.class })
+@Deploy("org.nuxeo.ecm.automation.core")
+public class CopyDocumentTest {
+
+    private static final String COPY_PATH = "/";
+
+    private static final String COPY_DOC_NAME = "copyDoc";
+
+    private static final String TARGET_PROPERTY_KEY = "target";
+
+    private static final String NAME_PROPERTY_KEY = "name";
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected AutomationService service;
+
+    @Inject
+    protected TransactionalFeature txFeature;
+
+    protected DocumentModel source;
+
+    protected OperationContext context;
+
+    @Before
+    public void setUp() {
+        source = session.createDocumentModel(COPY_PATH, "Source", "File");
+        source = session.createDocument(source);
+        session.save();
+
+        context = new OperationContext(session);
+    }
+
+    @After
+    public void tearDown() {
+        context.close();
+        session.removeChildren(session.getRootDocument().getRef());
+        session.save();
+    }
+
+    @Test
+    public void testLifeCycleResetOnCopy() throws OperationException {
+        String initialLifeCycleState = source.getCurrentLifeCycleState();
+
+        // Change the source current lifecycle state
+        source.followTransition("approve");
+        source = session.saveDocument(source);
+        txFeature.nextTransaction();
+
+        // Call Document.Copy without reseting the lifecycle state of the copy
+        Map<String, Serializable> params = new HashMap<>();
+        params.put(TARGET_PROPERTY_KEY, COPY_PATH);
+        params.put(NAME_PROPERTY_KEY, COPY_DOC_NAME);
+        params.put(CoreEventConstants.RESET_LIFECYCLE, Boolean.FALSE);
+
+        context.setInput(source);
+        DocumentModel result = (DocumentModel) service.run(context, CopyDocument.ID, params);
+        txFeature.nextTransaction();
+
+        result = session.getDocument(result.getRef());
+        assertEquals("approved", result.getCurrentLifeCycleState());
+
+        // Call Document.Copy and reset the lifecycle state of the copy
+        params = new HashMap<>();
+        params.put(TARGET_PROPERTY_KEY, COPY_PATH);
+        params.put(NAME_PROPERTY_KEY, COPY_DOC_NAME);
+        params.put(CoreEventConstants.RESET_LIFECYCLE, Boolean.TRUE);
```

Comment:
you don't need to create the whole map here, just `put` the new boolean value that you want

Reply:
Map.of gives you an immutable implementation of Map so I make 2 maps